### DISCUSSION TITLE
Firebase Cloud Messaging Token for Push Notifications

### DISCUSSION THREAD

**Discussion by kpespisa** [shinyorg/shiny#1323] [Q&A]
I've setup Shiny.Push v3 in my iOS app and configured it to work with Firebase. The registration token I get from the push manager is the APNS token, not the Firebase token. I'm wondering if that's intended?

There's an easy solution to get the correct one:

    public async Task CheckPermission()
    {
        var push = Host.Current.Services.GetService<IPushManager>();
        Logger.Write("FCM Token: " + push.RegistrationToken);  // this is APNS token

        var fcmToken = await Messaging.SharedInstance.FetchTokenAsync();
        Logger.Write("FCM Token: " + fcmToken);  // this one is from Firebase
    }

**Comment by aritchie** [comment]
Please file potential issues with proper details.  This area is meant for "how tos"
